### ISSUE TITLE
MNIST doesn't work on Mac mini m1

### ISSUE THREAD

**Issue by DongqiShen** [pytorch/examples#915]
I run it on my Mac mini m1, but the loss is always nan. 
Then, I tried it on the Ubuntu and It worked.
It seems that pytorch doesn't work well on m1.

**Comment by xshaun** [comment]
Is the random seed same one in your settings? It would be helpful if you have a try on other models and give more runtime information.

**Comment by shreyaspadhy** [comment]
Hi Dongqi, did you have any followups to this issue? I am running into a closely related issue on an M1 Mac Mini where the default MNIST example does not seem to have good test time evaluation (accuracy is always chance ~10%), but the training loss goes down a significant amount. However, on a 2017 x86 Macbook Pro, the test accuracy reaches ~99% within 2 epochs, with similar training loss as the M1 machine. I'm happy to provide any further information to get to the bottom of this. 

On both machines, I set up a Python, Conda and Pytorch Environment from scratch, following this tutorial pretty much exactly - https://towardsdatascience.com/yes-you-can-run-pytorch-natively-on-m1-macbooks-and-heres-how-35d2eaa07a83

I'm on Python 3.9.6 on both machines.

**Comment by msaroufim** [comment]
Super strange, that said for M1 updates I'd suggest following this thread instead https://github.com/pytorch/pytorch/issues/47702#issuecomment-965625139
